packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
 pragma solidity >=0.7.6;
 
+import "./OVM_Testable.sol"; //TODO: replace this with the normal UMA Testable once we can use 0.8 solidity.
+
 import { OVM_CrossDomainEnabled } from "@eth-optimism/contracts/libraries/bridge/OVM_CrossDomainEnabled.sol";
+import { Lib_PredeployAddresses } from "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+
+// Define some interfaces and helper libraries. This is temporary until such that time that we can bump the solidity
+// version in these contracts to 0.8.x and import the rest of these libs from other UMA contracts in the rpo.
+library TokenHelper {
+    function safeTransferFrom(
+        address token,
+        address from,
+        address to,
+        uint256 value
+    ) internal {
+        // bytes4(keccak256(bytes('transferFrom(address,address,uint256)')));
+        (bool success, bytes memory data) = token.call(abi.encodeWithSelector(0x23b872dd, from, to, value));
+        require(
+            success && (data.length == 0 || abi.decode(data, (bool))),
+            "TokenHelper::transferFrom: transferFrom failed"
+        );
+    }
+}
+
+interface TokenLike {
+    function balanceOf(address guy) external returns (uint256 wad);
+}
+
+interface StandardBridgeLike {
+    function withdrawTo(
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _l1Gas,
+        bytes calldata _data
+    ) external;
+}
+
+/**
+ * @title OVM Bridge Deposit Box.
+ * @notice Accepts deposits on Optimism L2 to relay to Ethereum L1 as part of the UMA insured relayer system.
+ */
+
+contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled, OVM_Testable {
+    /*************************************
+     *  OVM DEPOSIT BOX DATA STRUCTURES  *
+     *************************************/
 
-contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled {
-    address public l1Owner;
+    address public l1WithdrawContract;
 
-    event Deposit(
+    bool public depositsEnabled = true;
+
+    // Track the total number of deposits. Used as a unique identifier for bridged transfers.
+    uint256 public numberOfDeposits;
+
+    struct WhitelistedToken {
+        address l1Token;
+        uint64 lastBridgeTime;
+    }
+
+    // Mapping of whitelisted L2Token to WhitelistedToken. Contains L1 TokenAddress and the last time this token type
+    // was bridged. Used to rate limit bridging actions to prevent DOS on L1.
+    mapping(address => WhitelistedToken) public whitelistedTokens;
+
+    // Minimum time that must elapse between bridging actions for a given token. Used to rate limit bridging back to L1.
+    uint64 public minimumBridgingDelay;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event L1WithdrawContractChanged(address oldL1WithdrawContract, address newL1WithdrawContract);
+    event TokenWhitelisted(address l1Token, address l2Token);
```

Comment:
Similar reasoning above, maybe rename this to `WhitelistToken`

Reply:
done